models/yang/sonic/sonic-port.yang

```diff
@@ -0,0 +1,79 @@
+module sonic-port {
+	namespace "http://github.com/Azure/sonic-port";
+	prefix prt;
+
+	import sonic-common {
+		prefix scommon;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC VLAN";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+
+	container sonic-port {
+
+		container PORT {
+
+			list PORT_LIST {
+				key "ifname";
+
+				leaf ifname {
+					type string {
+						pattern "Ethernet([1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[0-9])" {
+							error-message "Invalid interface name";
+							error-app-tag interface-name-invalid;
+						}
+					}
+				}
+
+				leaf index {
```

Comment:
Checkout & mark fields that are mandatory.

Reply:
Done